x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/xsearch/action/XSearchAction.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.xsearch.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class XSearchAction extends ActionType<SearchResponse> {
+
+    public static final XSearchAction INSTANCE = new XSearchAction();
+
+    static final String NAME = "indices:data/read/xsearch";
+
+    private XSearchAction() {
+        super(NAME, SearchResponse::new);
+    }
+
+    public static class Request extends ActionRequest implements IndicesRequest.Replaceable {
+
+        private String[] indices;
+
+        private final String query;
+
+        private final boolean explain;
+        private IndicesOptions indicesOptions = IndicesOptions.fromOptions(false, false, true, false, false, true, false, false);
```

Comment:
Should we use one of the existing static methods from IndicesOptions class, like `IndicesOptions.strictExpandOpenAndForbidClosedIgnoreThrottled()`

Reply:
Oh, nice, I hadn't noticed those, thanks for the suggestion! 👍